packages/web-components/@groupby/elements-products/src/products-sayt.ts

```diff
@@ -15,13 +17,26 @@ import { ProductsBase } from '.';
  */
 @customElement('gbe-products-sayt')
 export default class ProductsSayt extends ProductsBase {
+  /**
+   * A random string suitable for use in stable IDs related to this
+   * component.
+   */
+  protected componentId = shortid.generate();
+
+  /**
+   * A string intended to be used as the name of the return event in
+   * cache requests for this component.
+   */
+  protected cacheResponseEventName = this.getCacheResponseEventName('products-sayt', this.componentId);
+
   /**
    * Binds relevant methods.
    */
   constructor() {
     super();
 
     this.setProductsFromEvent = this.setProductsFromEvent.bind(this);
```

Comment:
To make this name less ambiguous, consider renaming this function to `setProductsFromProductsEvent` so it can't be confused with `setProductsFromCacheEvent`.